services/appflowy-collaborate/src/indexer/indexer_scheduler.rs

```diff
@@ -159,8 +163,15 @@ impl IndexerScheduler {
       return Ok(());
     }
 
-    let embedder = self.create_embedder()?;
     let indexed_collab = indexed_collab.into();
+    if !self.is_indexing_enabled(&indexed_collab.collab_type) {
```

Comment:
You can use `IndexerProvider::indexer_for` method which already return `None` if indexing is not enabled. This way you also don't need to copy indexer provider inside of rayon thread.